Add RootHash to L1BlockInfo

Inkvi opened the pull request:
- The change is necessary for IBC clients so they can verify consensus updates

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **New Features**
  - Enhanced block information tracking with the addition of a new RootHash field to improve data integrity and verification processes.

- **Improvements**
  - Updated data serialization and deserialization methods to accommodate the new RootHash field, ensuring seamless integration with existing systems.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

dshiell commented:
@coderabbitai review

dshiell reviewed (commented):
Looks like one possible coderabbit issue?

nicopernas commented:
Please change the base to the `polymer/v1.2.0-dev` branch.

Inkvi commented:
> Please change the base to the `polymer/v1.2.0-dev` branch.

There is no branch like that in the repo. Did you mean `polymer/v1.2.0`?

Inkvi commented on op-node/rollup/derive/l1_block_info.go:
```diff
@@ -90,6 +92,9 @@ func (info *L1BlockInfo) MarshalBinary() ([]byte, error) {
 	if err := solabi.WriteEthBytes32(w, info.L1FeeScalar); err != nil {
 		return nil, err
 	}
+	if err := solabi.WriteHash(w, info.RootHash); err != nil {
+		return nil, err
+	}
 	return w.Bytes(), nil
 }
 
```
Good catch @coderabbitai!
Update L1InfoArguments from 8 to 9.

dshiell commented:
> > Please change the base to the `polymer/v1.2.0-dev` branch.
> 
> There is no branch like that in the repo. Did you mean `polymer/v1.2.0`?

Yeah I'm 99% sure that's the correct one to merge with. @nicopernas can confirm.

nicopernas commented:
> > > Please change the base to the `polymer/v1.2.0-dev` branch.
> 
> > 
> 
> > There is no branch like that in the repo. Did you mean `polymer/v1.2.0`?
> 
> 
> 
> Yeah I'm 99% sure that's the correct one (polymer/v1.2.0) to merge with. @nicopernas can confirm.

Yes! I meant `polymer/v1.2.0`. I messed up :)

Inkvi commented:
> > > > Please change the base to the `polymer/v1.2.0-dev` branch.
> > 
> > 
> > > 
> > 
> > 
> > > There is no branch like that in the repo. Did you mean `polymer/v1.2.0`?
> > 
> > 
> > Yeah I'm 99% sure that's the correct one (polymer/v1.2.0) to merge with. @nicopernas can confirm.
> 
> Yes! I meant `polymer/v1.2.0`. I messed up :)

I changed the base of the PR and rebased the changes on top of that branch

jlafiandra6 reviewed (approved):
LGTM
